### PR TITLE
Enable e2e tests in CI

### PR DESCRIPTION
**Pull request by yiqian1**

### CODE REVIEW

**Review by whchung — CHANGES_REQUESTED** [review]
Please break e2e tests to smaller ones to leverage multi-core CPUs we have on CI nodes.

Long CI jobs also reduce iterations we can do per day.

Ideally, I want every CI job be concluded within 20 minutes.

**Comment by whchung** [comment]
@yiqian1 build 2942 for this job took 27 minutes to finish:

http://ml-ci.amd.com:21096/job/MLIR/job/MLIR-downstream-unit-test/2942/

Please help check existing e2e tests so the job can be completed within 20 minutes.


Meanwhile, we likely need a standalone nightly CI job to validate all configs. We'll deal with that in a couple of weeks after we put more fix in the code generation process.

File: mlir/test/mlir-miopen-driver/e2e/conv2d_host_validation_kcyx_nchw_nkhw.mlir [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// COM: kcyx/nchw/nkhw
```
Comment (jerryyin) [inline mlir/test/mlir-miopen-driver/e2e/conv2d_host_validation_kcyx_nchw_nkhw.mlir:1]: What does this line do?

File: mlir/test/mlir-miopen-driver/e2e/conv2d_host_validation_yxck_nhwc_nhwk1.mmlir [hunk 2]
```diff
@@ -0,0 +1,23 @@
+// COM: yxck/nhwc/nhwk
```
Comment (jerryyin) [inline mlir/test/mlir-miopen-driver/e2e/conv2d_host_validation_yxck_nhwc_nhwk1.mmlir]: Wrong filename: `.mlir` instead of `.mmlir`

File: mlir/test/mlir-miopen-driver/e2e/conv2d_host_validation_yxck_nhwc_nhwk2.mlir [hunk 3]
```diff
@@ -0,0 +1,20 @@
+// COM: yxck/nhwc/nhwk
```
Comment (jerryyin) [inline mlir/test/mlir-miopen-driver/e2e/conv2d_host_validation_yxck_nhwc_nhwk2.mlir:1]: Could you summarize the difference and similarities between:
- conv2d_host_validation_yxck_nhwc_nhwk1.mlir 
- conv2d_host_validation_yxck_nhwc_nhwk2.mlir 

Such that it becomes more apparent what you are testing by looking at the filename

**Review by jerryyin — CHANGES_REQUESTED** [review]

**Comment by whchung** [comment]
@yiqian1 the new commit doesn't help reduce total test time.

build 2942:
http://ml-ci.amd.com:21096/job/MLIR/job/MLIR-downstream-unit-test/2942/consoleFull
```
Testing Time: 737.27s
  Unsupported      :  41
  Passed           : 616
  Expectedly Failed:   2
```

build 2943:
http://ml-ci.amd.com:21096/job/MLIR/job/MLIR-downstream-unit-test/2943/consoleFull
```
Testing Time: 739.74s
  Unsupported      :  41
  Passed           : 619
  Expectedly Failed:   2
```

**Comment by whchung** [comment]
@yiqian1 what is the x86 ISA we get right now? are they compiled with OptLevel=3?

**Comment by whchung** [comment]
@yiqian1 as CPU convolution logic takes tremendous amount of time now, need to pay close attention on how the x86 codes get produced. please check how we set `clOptLevel` in `lib/ExecutionEngine/JitRunner.cpp`.

**Comment by whchung** [comment]
For example, when we build GPU kernels, we make sure we use `-O3` to keep LLVM AMDGPU backend happy:

```
    llvm::PassManagerBuilder builder;
    builder.OptLevel = 3;
```

File: mlir/test/mlir-miopen-driver/e2e/conv2d_host_validation_kcyx_nchw_nkhw.mlir (hunk 1 above)
Comment (yiqian1) [reply mlir/test/mlir-miopen-driver/e2e/conv2d_host_validation_kcyx_nchw_nkhw.mlir:1]: It's a comment.

File: mlir/test/mlir-miopen-driver/e2e/conv2d_host_validation_yxck_nhwc_nhwk1.mmlir (hunk 2 above)
Comment (yiqian1) [reply mlir/test/mlir-miopen-driver/e2e/conv2d_host_validation_yxck_nhwc_nhwk1.mmlir]: Oops

File: mlir/lib/ExecutionEngine/JitRunner.cpp [hunk 4]
```diff
@@ -71,16 +71,16 @@ struct Options {
...
-                            llvm::cl::cat(optFlags)};
+                            llvm::cl::cat(optFlags), llvm::cl::init(false)};
   llvm::cl::opt<bool> optO2{"O2",
                             llvm::cl::desc("Run opt passes and codegen at O2"),
-                            llvm::cl::cat(optFlags)};
+                            llvm::cl::cat(optFlags), llvm::cl::init(false)};
   llvm::cl::opt<bool> optO3{"O3",
                             llvm::cl::desc("Run opt passes and codegen at O3"),
-                            llvm::cl::cat(optFlags)};
+                            llvm::cl::cat(optFlags), llvm::cl::init(true)};
```
Comment (whchung) [inline mlir/lib/ExecutionEngine/JitRunner.cpp:83]: @yiqian1 try not change the default value. Set it when we initialize `JitRunner` in the client application.

**Comment by whchung** [comment]
@yiqian1 CPU convolution time is still not reduced. I would advise you benchmark those configs individually to make sure we do get optimized x86 ISA or not.

```
Testing Time: 748.06s
  Unsupported      :  41
  Passed           : 620
  Expectedly Failed:   2
```

File: mlir/lib/ExecutionEngine/JitRunner.cpp (hunk 4 above)
Comment (yiqian1) [reply mlir/lib/ExecutionEngine/JitRunner.cpp:83]: It doesn't help anyway.

**Comment by whchung** [comment]
Closing this PR as it's filed prematurely.
